src/vacuum-card.js

```diff
@@ -208,9 +242,11 @@ class VacuumCard extends LitElement {
         @click="${(e) => e.stopPropagation()}"
       >
         <paper-button slot="dropdown-trigger">
-          <ha-icon icon="mdi:fan"></ha-icon>
+          <ha-icon icon="mdi:${`${icon}`}"></ha-icon>
           <span show=${true}>
-            ${localize(`source.${source}`) || source}
+            ${mopSetting
+              ? `${mopSetting}`
+              : localize(`source.${source}`) || `${source}`}
```

Comment:
My vacuum is mopping and vacuuming at the same time. In this case, either speed or mopSettings is displayed.